chore: adjust PR template to match microdelivery specification (#3060)
kyma-project/busola: .github/pull-request-template.md

<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines, templates, the recommended Git workflow, and any related documentation.
2. Read and submit the required Contributor Licence Agreements (https://github.com/kyma-project/community/blob/main/CONTRIBUTING.md#agreements-and-licenses).
3. Test your changes and attach their results to the pull request.
4. Update the relevant documentation.

> If the pull request requires a decision, follow the [decision-making process](https://github.com/kyma-project/community/blob/main/governance.md) and replace the PR template with the [decision record template](https://github.com/kyma-project/community/blob/main/.github/ISSUE_TEMPLATE/decision-record.md).

Reviewer: ```suggestion
If the pull request requires a decision, follow the [decision-making process](https://github.com/kyma-project/community/blob/main/docs/governance/01-governance.md) and replace the PR's template with the [decision record template](https://github.com/kyma-project/community/blob/main/.github/ISSUE_TEMPLATE/decision-record.md).
```
the first link doesn't work - I guess you meant this one but please check it

**Related issue(s)**

<!-- If you refer to a particular issue, provide its number. For example, `Resolves #123`, `Fixes #43`, or `See also #33`. -->

> **Confirm that everything was done**

Reviewer: what about: `Confirm that your PR meets all requirements`
or, even more simply, why not call it `Definition of done`

> - [ ] PR title starts with prefix

Reviewer: ```suggestion
- [ ] The PR's title starts with one of the following prefixes:

  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests

>   - chore: Maintain. Changes to the build process or auxiliary tools/libraries/workflows etc.

Reviewer: ```suggestion
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
```

> - [ ] Related issues are linked (for links to internal trackers you can use the issue ids like `backlog#4567`)

Reviewer: ```suggestion
- [ ] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
```

> - [ ] It's clear why is this PR was created and what it does

Reviewer: ```suggestion
- [ ] Explain clearly why you created the PR and what changes it introduces
```

> - [ ] All necessary steps are delivered (like e.g. tests, documentation, merging)

Reviewer: ```suggestion
- [ ] All necessary steps are delivered, for example, tests, documentation, merging
```
